protostar/configuration_file/configuration_file_v1.py

```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+from typing import List, Optional, Union
+
+from protostar.protostar_toml.io.protostar_toml_reader import ProtostarTOMLReader
+from protostar.utils.protostar_directory import VersionManager, VersionType
+
+from .configuration_file import (
+    ConfigurationFile,
+    ContractNameNotFoundException,
+    PrimitiveTypesSupportedByConfigurationFile,
+)
+
+
+class ConfigurationFileV1(ConfigurationFile):
+    def __init__(
+        self, protostar_toml_reader: ProtostarTOMLReader, project_root_path: Path
```

Comment:
I think these classes should **own and construct** `ProtostarTOMLReader` class by themselves, so that these could be replaceable by future versions of `ConfigurationFile`.

Reply:
Disagree. Dependencies (constructor args) aren't part of the ConfigurationFile interface.

Reply:
My thing is, `ProtostarTOMLReader` is **not** an external dependency of `ConfigurationFile` but internal one.

If we will hypothetically introduce `ProtostarJSONReader` in the future, with an intent of replacing `protostar.toml` with `protostar.json`; it will be natural to introduce `ConfigurationFileV3`, but having the TOML reader as an external dependency passed from outside will make all of this tricky.

Conceptually, nobody outside `ConfigurationFile` should know that `ProtostarTOMLReader` is used internally. It should be a hidden implementation detail.

Reply:
(Internal discussion ended without consensus)

I'm sure I can make it work in a nice loosely coupled. I'll probably create big PR though.